### ISSUE TITLE
Primes in mixed characteristic

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#2611]
Another easy computation that does not work over the integers:
```m2
R = ZZ[e_1, e_2, e_3, Degrees =>{1,2,3}]
I = ideal(2, e_1*e_2 - e_3)
isPrime I -- error: no applicable strategy for (minimalPrimes,Ideal)
```
I believe the algorithm of Gianni-Trager-Zacharias should work for `minimalPrimes` over integers. Is that right, @jchen419?

Somewhat related, codimension also doesn't work, but I believe it is at least a well-defined notion, since we can just look at the primary decomposition of I.
```m2
codim I -- error: codim: expected an affine ring (consider Generic=>true to work over QQ)
```

Reported by @sashapevzner

**Comment by jchen419** [comment]
Yes, in general the algorithms described in GTZ work over any PID (at least if factorization is algorithmic). So using `MinimalPrimes.m2` it should in theory be possible to test primality in polynomial rings over `ZZ`.

I agree that codimension - which is unambiguously defined for every ideal in any commutative ring whatsoever - should be made to work over `ZZ`, as `Generic => true` is not really always an acceptable workaround.

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

All three still fail on 1.26.06, from the same places: `isPrime I` and `minimalPrimes I` raise
`no applicable strategy for (minimalPrimes,Ideal)` at `MinimalPrimes.m2:239`, and `codim I` raises
`expected an affine ring (consider Generic=>true to work over QQ)` at `monideal.m2:114`.

Worth making @jchen419's point about the workaround concrete, since the error message recommends it:

```m2
i : codim(I, Generic => true)

o = 4
```

`I` contains 2, so over `QQ` it is the unit ideal — the suggestion answers a question about a
different ideal rather than giving a usable value for this one.

The `codim` half also has a second open row that neither issue links to: #3557, where `rank` over
`ZZ` fails with this same error and there is no way to pass `Generic => true` through it at all. Same
gap, reached from a much more ordinary direction.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
